### PR TITLE
Fix SR-1444, Match Xcode's c99name

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
dependent on #336

### CODE REVIEW

File: Tests/Xcodeproj/FunctionalTests.swift [hunk 1]
```diff
@@ -21,13 +21,55 @@ class FunctionalTests: XCTestCase {
...
+}
+
+func write(path: String, write: (OutputByteStream) -> Void) throws -> String {
+    try fopen(path, mode: .Write) { fp in
+        let stream = OutputByteStream()
+        write(stream)
+        try fputs(stream.bytes.bytes, fp)
+    }
+    return path
+}
```
Comment (ddunbar) [inline Tests/Xcodeproj/FunctionalTests.swift]: Should this just go into `Utility` or somewhere?

File: Tests/Xcodeproj/FunctionalTests.swift [hunk 2]
```diff
@@ -21,13 +21,55 @@ class FunctionalTests: XCTestCase {
             XCTAssertXcodeprojGen(prefix)
             let pbx = Path.join(prefix, "Library.xcodeproj")
             XCTAssertDirectoryExists(pbx)
-            // FIXME: Move this to a method.
-            try popen(["env", "-u", "TOOLCHAINS", "xcodebuild", "-project", pbx])
+            XCTAssertXcodeBuild(project: pbx)
```
Comment (ddunbar) [inline Tests/Xcodeproj/FunctionalTests.swift]: I forget the details of how these fixtures are written, where is this actually building?

Comment (aciidb0mb3r) [inline Tests/Xcodeproj/FunctionalTests.swift]: the fixture is copied into a temp folder somewhere `/var/.../something/`
xcodebuild builds in a folder called "build" inside the copied fixture

File: Tests/Xcodeproj/FunctionalTests.swift (hunk 1 above)
Comment (aciidb0mb3r) [inline Tests/Xcodeproj/FunctionalTests.swift]: how about Utility/File.swift or Utility/misc.swift?

File: Tests/Xcodeproj/FunctionalTests.swift (hunk 2 above)
Comment (ddunbar) [inline Tests/Xcodeproj/FunctionalTests.swift]: Ok great

**Comment by aciidb0mb3r** [comment]
rebased

**Comment by ddunbar** [comment]
I see test failures with this:
```
output: Compile Swift Module 'Module_Name_1' (1 sources)
Compile Swift Module 'Module_Name_2' (1 sources)
/private/var/folders/yh/3pgbfsz165q2t6pyg3ktvw0r0000gn/T/Miscellaneous_Spaces Fixture.BdeFtu/Spaces Fixture/Sources/Module Name 2/Foo.swift:1:8: error: no such module 'ModuleName1'
import ModuleName1
       ^
<unknown>:0: error: build had 1 command failures
swift-build: error: exit(1): /Volumes/Data/ddunbar/public/swift-project/build/Ninja-ReleaseAssert/swiftpm-macosx-x86_64/debug/swift-build-tool -f /private/var/folders/yh/3pgbfsz165q2t6pyg3ktvw0r0000gn/T/Miscellaneous_Spaces\ Fixture.BdeFtu/Spaces\ Fixture/.build/debug.yaml

SWIFT_EXEC: nil
swift-build: /Volumes/Data/ddunbar/public/swift-project/build/Ninja-ReleaseAssert/swiftpm-macosx-x86_64/debug/swift-build
/Volumes/Data/ddunbar/public/swift-project/swiftpm/Tests/Functional/MiscellaneousTests.swift:371: error: -[FunctionalTestSuite.MiscellaneousTestCase testSpaces] : failed - `swift build -c Debug' failed:

exit(1): /Volumes/Data/ddunbar/public/swift-project/build/Ninja-ReleaseAssert/swiftpm-macosx-x86_64/debug/swift-build --chdir /var/folders/yh/3pgbfsz165q2t6pyg3ktvw0r0000gn/T/Miscellaneous_Spaces\ Fixture.BdeFtu/Spaces\ Fixture --configuration debug

    Building Release
output: Compile Swift Module 'Module_Name_1' (1 sources)
Compile Swift Module 'Module_Name_2' (1 sources)
/private/var/folders/yh/3pgbfsz165q2t6pyg3ktvw0r0000gn/T/Miscellaneous_Spaces Fixture.BdeFtu/Spaces Fixture/Sources/Module Name 2/Foo.swift:1:8: error: no such module 'ModuleName1'
import ModuleName1
       ^
<unknown>:0: error: build had 1 command failures
swift-build: error: exit(1): /Volumes/Data/ddunbar/public/swift-project/build/Ninja-ReleaseAssert/swiftpm-macosx-x86_64/debug/swift-build-tool -f /private/var/folders/yh/3pgbfsz165q2t6pyg3ktvw0r0000gn/T/Miscellaneous_Spaces\ Fixture.BdeFtu/Spaces\ Fixture/.build/release.yaml

SWIFT_EXEC: nil
swift-build: /Volumes/Data/ddunbar/public/swift-project/build/Ninja-ReleaseAssert/swiftpm-macosx-x86_64/debug/swift-build
/Volumes/Data/ddunbar/public/swift-project/swiftpm/Tests/Functional/MiscellaneousTests.swift:371: error: -[FunctionalTestSuite.MiscellaneousTestCase testSpaces] : failed - `swift build -c Release' failed:

exit(1): /Volumes/Data/ddunbar/public/swift-project/build/Ninja-ReleaseAssert/swiftpm-macosx-x86_64/debug/swift-build --chdir /var/folders/yh/3pgbfsz165q2t6pyg3ktvw0r0000gn/T/Miscellaneous_Spaces\ Fixture.BdeFtu/Spaces\ Fixture --configuration release

/Volumes/Data/ddunbar/public/swift-project/swiftpm/Tests/Functional/MiscellaneousTests.swift:372: error: -[FunctionalTestSuite.MiscellaneousTestCase testSpaces] : failed - Expected file doesn’t exist: /var/folders/yh/3pgbfsz165q2t6pyg3ktvw0r0000gn/T/Miscellaneous_Spaces Fixture.BdeFtu/Spaces Fixture/.build/debug/ModuleName1.build/Foo.swift.o
Test Case '-[FunctionalTestSuite.MiscellaneousTestCase testSpaces]' failed (0.733 seconds).
Test Case '-[FunctionalTestSuite.MiscellaneousTestCase testTestDependenciesComplex]' started.
Test Case '-[FunctionalTestSuite.MiscellaneousTestCase testTestDependenciesComplex]' passed (0.052 seconds).
Test Case '-[FunctionalTestSuite.MiscellaneousTestCase testTestDependenciesSimple]' started.
Test Case '-[FunctionalTestSuite.MiscellaneousTestCase testTestDependenciesSimple]' passed (0.000 seconds).
Test Case '-[FunctionalTestSuite.MiscellaneousTestCase testTipHasNoPackageSwift]' started.
    Building Debug
    Building Release
Test Case '-[FunctionalTestSuite.MiscellaneousTestCase testTipHasNoPackageSwift]' passed (5.033 seconds).
```

File: Sources/PackageType/c99name().swift [hunk 3]
```diff
@@ -178,9 +178,9 @@ public func c99name(name: String) throws  -> String {
             0x4E00...0x9FA5,
             // Hangul,
             0xAC00...0xD7A3:
-            return true
+            return $0
         default:
-            return false
+            return "_"
         }
     }
```
Comment (ddunbar) [inline Sources/PackageType/c99name().swift]: I think there is another change needed here, there is a loop below this one that deletes leading characters I believe.

**Comment by aciidb0mb3r** [comment]
@ddunbar Fixed

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by ddunbar** [comment]
Have you looked at the test failure? It does look like it might be related.

**Comment by aciidb0mb3r** [comment]
This should fix it https://github.com/apple/swift-integration-tests/pull/6

**Pull request merged**